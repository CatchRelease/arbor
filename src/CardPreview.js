import styled from 'react-emotion';
import PropTypes from 'prop-types';
import { ratio } from 'styled-system';

import Box from './Box';

const CardPreview = styled(Box.withComponent('div'))`
  background: url(${props => props.image}) center center no-repeat
    ${props => props.theme.colors.black};
  background-size: contain;
  ${ratio};
`;

CardPreview.propTypes = {
  image: PropTypes.string.isRequired,
  ...ratio.propType
};

CardPreview.defaultProps = {
  ratio: 9 / 16
};

export default CardPreview;
