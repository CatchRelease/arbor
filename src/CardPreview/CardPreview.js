import styled from '@emotion/styled';
import PropTypes from 'prop-types';
import { ratio } from 'styled-system';

import Box from '../Box';
import createWithComponent from '../utils/createWithComponent';

const CardPreview = styled(
  createWithComponent(Box, 'div', {
    defaultProps: {
      ratio: 9 / 16
    }
  })
)`
  background: url(${props => props.image}) center center no-repeat
    ${props => props.theme.colors.black};
  background-size: contain;
  ${ratio};
`;

CardPreview.propTypes = {
  image: PropTypes.string.isRequired,
  ...ratio.propType
};

export default CardPreview;
