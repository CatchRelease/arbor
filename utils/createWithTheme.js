import React from 'react';
import { CacheProvider, ThemeProvider } from '@emotion/react';
import createCache from '@emotion/cache';
import { create } from 'react-test-renderer';
import theme from '../src/theme';

const myCache = createCache({ key: 'test' });
myCache.compat = true;

export default (Component) =>
  create(
    <CacheProvider value={myCache}>
      <ThemeProvider theme={theme}>{Component}</ThemeProvider>
    </CacheProvider>
  );
