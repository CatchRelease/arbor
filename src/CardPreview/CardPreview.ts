import styled from '@emotion/styled';

import { ratio } from '../utils';

import Box from '../Box';

type Props = {
  image: string;
  ratio?: number;
};

const CardPreview = styled(Box)<Props>`
  background: url(${(props) => props.image}) center center no-repeat
    ${(props) => props.theme.colors.monochrome.black};
  background-size: contain;
  ${ratio};
`;

CardPreview.defaultProps = {
  ratio: 9 / 16
};

export default CardPreview;
