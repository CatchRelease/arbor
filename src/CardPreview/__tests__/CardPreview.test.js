import createWithTheme from '../../../utils/createWithTheme';

import CardPreview from '../CardPreview';

describe('<CardPreview />', () => {
  it('renders a CardPreview properly', () => {
    const tree = createWithTheme(<CardPreview image="example.gif" />);

    expect(tree).toMatchSnapshot();
  });
});
