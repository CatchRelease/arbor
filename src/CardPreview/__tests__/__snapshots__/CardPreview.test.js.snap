// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CardPreview /> renders a CardPreview properly 1`] = `
.emotion-0 {
  font-family: Motiva Sans;
  box-sizing: border-box;
  background: url(example.gif) center center no-repeat #000000;
  -webkit-background-size: contain;
  background-size: contain;
  height: 0;
  padding-bottom: 56.25%;
}

<div
  className="emotion-0 emotion-1"
/>
`;
